qa/tasks/salt_manager.py

```diff
@@ -0,0 +1,155 @@
+'''
+Salt "manager" module
+
+Usage: First, ensure that there is a role whose name corresponds
+to the value of the master_role variable, below. Second, in your
+task, instantiate a SaltManager object:
+
+    from salt_manager import SaltManager
+
+    sm = SaltManager(ctx, config)
+
+Third, enjoy the SaltManager goodness - e.g.:
+
+    sm.ping_minions()
+
+'''
+import logging
+import re
+
+from cStringIO import StringIO
+from teuthology import misc
+from teuthology.contextutil import safe_while
+from teuthology.exceptions import (CommandFailedError, ConfigError)
+from teuthology.orchestra.remote import Remote
+from teuthology.orchestra import run
+from util import get_remote_for_role
+
+log = logging.getLogger(__name__)
+master_role = 'client.salt_master'
+
+
+class SaltManager(object):
+
+    def __init__(self, ctx, config):
+        config["master_remote"] = get_remote_for_role(ctx, master_role).name
+        self.config = config
+        self.ctx = ctx
+        self.cluster = ctx.cluster
+        self.master_remote = Remote(self.config.get('master_remote'))
+        self.remotes = ctx.cluster.remotes
+
+    def __systemctl_cluster(self, subcommand=None, service=None):
+        """
+        Do something to a systemd service unit on all remotes (test nodes) at
+        once
+        """
+        self.cluster.run(args=[
+            'sudo', 'systemctl', subcommand, '{}.service'.format(service)])
+
+    def __systemctl_remote(self, remote, subcommand=None, service=None):
+        """
+        Do something to a systemd service unit on a single remote (test node)
+        """
+        try:
+            remote.run(args=[
+                'sudo', 'systemctl', subcommand, '{}.service'.format(service)])
+        except CommandFailedError:
+            log.warning("Failed to {} {}.service!".format(subcommand, service))
+            remote.run(args=[
+                'sudo', 'systemctl', 'status', '--full', '--lines=50',
```

Comment:
Hm, don't remember why I put `--lines=50` here. The idea was to do `system status --full` without limiting the number of lines (?)

Reply:
`--full` prevents ellipsizing/truncation, and `--lines=50` causes more lines to be emitted (default is something like 10 or 20), so this is OK.